File: solutions/fortran/leap/3/leap.f90
module leap
  implicit none

contains

  logical function is_leap_year(year)
    integer :: year
    
    is_leap_year = (iand(year, 3) == 0) .and. ((mod(year,25) /= 0 ) .or.  (iand(year, 15) == 0))
  end function

end module

File: solutions/free-pascal/leap/1/Leap.pas
unit Leap;

{$mode ObjFPC}{$H+}

interface

function LeapYear(const year : integer) : boolean;

implementation

uses SysUtils;

function LeapYear(const year : integer) : boolean;
begin

  result := (year mod 4 = 0) and ((year mod 16 = 0) or (year mod 25 <> 0));
 

end;

end.
